core/runtime/src/main/java/io/quarkus/runtime/logging/LoggingSetupRecorder.java

```diff
@@ -106,44 +114,54 @@ public void initializeLogging(LogConfig config, LogBuildTimeConfig buildConfig,
-            additionalNamedHandler.setFilter(new LogCleanupFilter(filterElements));
-        }
-        namedHandlers.putAll(additionalNamedHandlersMap);
-
-        for (Map.Entry<String, CategoryConfig> entry : categories.entrySet()) {
-            final CategoryBuildTimeConfig buildCategory = isSubsetOf(entry.getKey(), buildConfig.categories);
-            final Level logLevel = getLogLevel(entry.getKey(), entry.getValue(), categories, buildConfig.minLevel);
-            final Level minLogLevel = buildCategory == null
-                    ? buildConfig.minLevel
-                    : buildCategory.minLevel.getLevel();
-
-            if (logLevel.intValue() < minLogLevel.intValue()) {
-                log.warnf("Log level %s for category '%s' set below minimum logging level %s, promoting it to %s", logLevel,
-                        entry.getKey(), minLogLevel, minLogLevel);
+        if (!categories.isEmpty()) {
+            Map<String, Handler> namedHandlers = createNamedHandlers(config, possibleFormatters, errorManager, filterElements);
 
-                entry.getValue().level = InheritableLevel.of(minLogLevel.toString());
+            Map<String, Handler> additionalNamedHandlersMap;
+            if (additionalNamedHandlers.isEmpty()) {
+                additionalNamedHandlersMap = Collections.emptyMap();
+            } else {
+                additionalNamedHandlersMap = new HashMap<>();
+                for (RuntimeValue<Map<String, Handler>> runtimeValue : additionalNamedHandlers) {
+                    runtimeValue.getValue().forEach(new BiConsumer<String, Handler>() {
+                        @Override
+                        public void accept(String s, Handler handler) {
+                            Handler previous = additionalNamedHandlersMap.putIfAbsent(s, handler);
+                            if (previous != null) {
+                                throw new IllegalStateException(String.format(
+                                        "Duplicate key %s (attempted merging values %s and %s)",
+                                        s, previous, handler));
+                            }
+                        }
+                    });
+                }
             }
-        }
 
-        for (Map.Entry<String, CategoryConfig> entry : categories.entrySet()) {
-            final String name = entry.getKey();
-            final Logger categoryLogger = logContext.getLogger(name);
-            final CategoryConfig categoryConfig = entry.getValue();
-            if (!categoryConfig.level.isInherited()) {
-                categoryLogger.setLevel(categoryConfig.level.getLevel());
-            }
-            categoryLogger.setUseParentHandlers(categoryConfig.useParentHandlers);
-            if (categoryConfig.handlers.isPresent()) {
-                addNamedHandlersToCategory(categoryConfig, namedHandlers, categoryLogger, errorManager);
+            for (Handler additionalNamedHandler : additionalNamedHandlersMap.values()) {
```

Comment:
since the patch is about nit-picking :P    a Map#values() should really be avoided.

Reply:
Indeed! I missed that one, thanks